src/main/webapp/app/components/commentIcon/CommentIcon.tsx

```diff
@@ -0,0 +1,245 @@
+    }, CLOSE_TOOLTIP_DURATION_MILLISECONDS);
+
+    timeoutId.current = id;
+  }
+
+  return (
+    <div onMouseEnter={handleMouseEnter} onMouseLeave={handleMouseLeave}>
+      <DefaultTooltip
+        overlayStyle={{ opacity: 1 }}
+        overlayInnerStyle={{ minWidth: '400px', padding: 0, borderColor: 'secondary' }}
+        placement="left"
+        destroyTooltipOnHide
+        visible={props.id === openCommentsIdStore.get()}
+        overlay={
+          <CommentBox
+            openCommentId={props.id}
+            comments={props.comments}
+            onCreateComment={props.onCreateComment}
+            onResolveComment={props.onResolveComment}
+            onUnresolveComment={props.onUnresolveComment}
+            onDeleteComments={props.onDeleteComments}
+          />
+        }
+      >
+        <div>
+          <FontAwesomeIcon size={props.size || 'lg'} id={props.id} icon={faComment} color={color} />
+        </div>
+      </DefaultTooltip>
+    </div>
+  );
+});
+
+export interface ICommentBoxProps {
+  comments: Comment[];
+  openCommentId: string;
+  onCreateComment: (content: string) => void;
+  onResolveComment: (index: number) => void;
+  onUnresolveComment: (index: number) => void;
+  onDeleteComments: (indices: number[]) => void;
+}
+
+const CommentBox = observer((props: ICommentBoxProps) => {
+  const scrollContainerRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    const scrollPosition = openCommentScrollPosition.get();
+    if (scrollPosition && scrollContainerRef.current) {
+      scrollContainerRef.current.scrollTop = scrollPosition;
+    }
+  });
+
+  useEffect(() => {
+    return () => {
+      if (props.comments && openCommentsIdStore.get() !== props.openCommentId) {
+        props.onDeleteComments(commentsToDeleteStore.commentIndiciesToDelete);
+        runInAction(() => commentsToDeleteStore.clear());
+      }
+    };
+  });
+
+  const sortedCommentItems = props.comments
+    ?.map((comment, index) => {
+      return [
+        comment.date,
+        <CommentItem
+          key={`${comment.date}-${comment.email}-${comment.content}`}
+          comment={comment}
+          index={index}
+          onResolveComment={() => {
+            runInAction(() => openCommentScrollPosition.set(scrollContainerRef.current.scrollTop));
+            props.onResolveComment(index);
+          }}
+          onUnresolveComment={() => {
+            runInAction(() => openCommentScrollPosition.set(scrollContainerRef.current.scrollTop));
+            props.onUnresolveComment(index);
+          }}
+        />,
+      ];
+    })
+    .sort((comment1, comment2) => {
+      return parseInt(comment2[0] as string, 10) - parseInt(comment1[0] as string, 10);
+    })
+    .map(comment => comment[1]);
+
+  return (
+    <>
+      {sortedCommentItems && sortedCommentItems.length > 0 && (
+        <div ref={scrollContainerRef} style={{ maxHeight: '320px', overflow: 'auto' }}>
+          <ListGroup className="ml-3 mr-3 mb-3 mt-3" flush>
```

Comment:
I think you can do this with bootstrap to specific all sides? Not 100% certain
```suggestion
          <ListGroup className="m-3" flush>
```

Reply:
You're right. Thanks for catching this